src/prefect/engine/task_runner.py

```diff
@@ -582,20 +582,20 @@ def get_task_inputs(
             # construct task inputs
             if edge.key is not None:
                 handlers[edge.key] = handler = getattr(
-                    edge.upstream_task, "result_handler", None
+                    edge.upstream_task, "result_handler", None # todo: here
```

Comment:
These are stored in a `handlers` dictionary so that we can easily access upstream result handlers down below, outside of this iteration